release/release_guide.md

```diff
@@ -0,0 +1,497 @@
+
+If there are no issues, reply on the vote thread to close the voting. Then, tally the votes in a separate email. Here’s an email template; please adjust as you see fit.
+
+```
+From: Release Manager
+
+To: dev@xtable.apache.org
+
+Subject: [RESULT] [VOTE] Release 1.2.3, release candidate #3
+
+ 
+
+I'm happy to announce that we have unanimously approved this release.
+
+ 
+
+There are XXX approving votes, XXX of which are binding:
+
+* approver 1
+
+* approver 2
+
+* approver 3
+
+* approver 4
+
+ 
+
+There are no disapproving votes.
+
+ 
+
+Thanks everyone!
+```
+
+## Checklist to proceed to the finalization step
+
+Community votes to release the proposed candidate, with at least three approving PMC votes
+
+# Fix any issues
+
+Any issues identified during the community review and vote should be fixed in this step.
+
+Code changes should be proposed as standard pull requests to the master branch and reviewed using the normal contributing process. Then, relevant changes should be cherry-picked into the release branch. The cherry-pick commits should then be proposed as the pull requests against the release branch, again reviewed and merged using the normal contributing process.
+
+Once all issues have been resolved, you should go back and build a new release candidate with these changes.
+
+## Checklist to proceed to the next step
+
+Issues identified during vote have been resolved, with fixes committed to the release branch.
+
+# Finalize the release
+
+Once the release candidate has been reviewed and approved by the community, the release should be finalized. This involves the final deployment of the release candidate to the release repositories, merging of the website changes, etc.
+
+## Current follow the below steps to finalize the release.
+
+1. Drop all RC orgapachextable-XXX in [Apache Nexus Staging Repositories](https://repository.apache.org/#stagingRepositories).
+2. change the version from ${RELEASE_VERSION}-rc${RC_NUM} to ${RELEASE_VERSION} against release branch, use command `mvn versions:set -DnewVersion=${RELEASE_VERSION}`, e.g. change 0.5.1-rc1 to 0.5.1.
+3. Commit and push the version change to release branch.
+    1. git commit -am "[MINOR] Update release version to reflect published version  ${RELEASE_VERSION}"
+    2. git push origin release-${RELEASE_VERSION}
+4. Repeat the steps from **_Generate Source Release (f) to Stage source releases on [dist.apache.org](http://dist.apache.org/) (i)_**. Including staging jars with the release version and uploading source release.
+   > **Note that make sure remove the -rc${RC_NUM} suffix when repeat the above steps. and please also verify the steps.  Ensure git tag is also done without -rc${RC_NUM}**
```

Comment:
@zabetak Regarding this comment https://github.com/apache/incubator-xtable/issues/486#issuecomment-2286943028, I have defined this a separate step after the release approval in the current guide.

Reply:
https://github.com/apache/incubator-xtable/issues/486#issuecomment-2288290391 
Discussed it here.